packages/web-components/fast-foundation/src/design-system-provider/design-system-provider.ts

```diff
@@ -255,42 +284,39 @@ export class DesignSystemProvider extends FASTElement
      */
     public disconnectedRegistry: Array<(provider: DesignSystemProvider) => void> | void;
 
-    /**
-     * The target of CSSCustomPropertyDefinitions registered
-     * with the provider. This will be #1 when adoptedStyleSheets are supported
-     * and #2 when they are not.
-     *
-     * 1. The `style` property of a CSSStyleRule on an adoptedStyleSheet
-     * 2. The `style` property of the element, resulting in inline styles
-     */
-    private customPropertyTarget: CSSStyleDeclaration;
-
     /**
      * Handle changes to design-system-provider IDL and content attributes
      * that reflect to design-system properties.
      */
     private attributeChangeHandler = {
         handleChange: (source: this, key: string) => {
             const value = this[key];
+            const manager = this.customPropertyManager;
 
             if (this.isValidDesignSystemValue(value)) {
                 this.designSystem[key] = value;
                 const property = this.designSystemProperties[key];
 
-                if (property && property.cssCustomProperty) {
-                    this.setCustomProperty({
+                if (property && property.cssCustomProperty && manager) {
+                    manager.set({
                         name: property.cssCustomProperty,
                         value,
                     });
                 }
             } else {
                 this.syncDesignSystemWithProvider();
                 const property = this.designSystemProperties[key].cssCustomProperty;
-                if (typeof property === "string") {
-                    this.deleteCustomProperty(property);
-                }
 
-                this.writeCustomProperties();
+                if (manager) {
+                    if (typeof property === "string") {
+                        manager.set({
+                            name: property,
+                            value: "",
```

Comment:
Empty string as a value?

Reply:
The intent is to remove it, but I should probably just add a public `delete` method.